openapi3filter/validation_error_encoder.go

```diff
@@ -148,7 +148,7 @@ func convertSchemaError(e *RequestError, innerErr *openapi3.SchemaError) *Valida
 	}
 
 	// Add error source
-	if e.Parameter != nil && e.Parameter.In == "query" {
```

Comment:
Why is this needed?

Reply:
This was the other issue I found. There were no tests for parameters in other locations. When I added a test for one in the header the source was not getting set.